0009-comit-btsieve-db.adoc

```diff
@@ -0,0 +1,98 @@
+= Persistent database for comit_node and btsieve
+Thomas Eizinger <thomas@coblox.tech>;
+:toc:
+:revdate: 2019-07-12
+
+NOTE: Author: {authors} +
+Date: {revdate} +
+Tracking issue: https://github.com/comit-network/comit-rs/issues/1060[#1060]
+
+== Context
+
+Currently, `comit_node` as-well as `btsieve` store all of their data just in-memory, which means it is lost as soon as the process stops.
+This is obviously not acceptable for software that deals with money and where the state being stored (transient keypairs, pre-images, etc) is crucial in accessing it.
+
+== Goals
+
+1. No separate process
+
+One of the core development goals of COMIT is to be as easily accessible as possible.
+Hence, any solution that involves running a _separate_ process for persistently storing data is not considered.
+All the solutions proposed here need to be embeddable into `comit_node` and `btsieve`.
+
+2. Atomic access
+
+Data integrity is key to our software.
+Hence, all access to the data needs be done in an atomic fashion.
+
+== Approach
+
+Given the above goals, we need to find an embeddable Rust database engine that supports atomic reads and writes.
+The research will therefore focus on checking the ecosystem and making a recommendation for one of the surveyed libraries.
+
+== Research
+
+Research resulted in the following database engines to be found:
+
+=== seld
+
+https://crates.io/crates/sled/0.24.1
+
+- + Written in pure Rust
+- + Actively developed
+
+- - alpha state (not yet 1.0, changing binary format between versions)
+
+=== rocksdb
+
+https://crates.io/crates/rocksdb/0.12.2
+
+- + RocksDB is developed by Facebook
+- + Popular (Parity uses it for substrate f.e.)
+
+- - Written C++ with Rust bindings
```

Comment:
I assume you are saying the rust bindings are available?

Reply:
The link to the crate are the Rust bindings.